### PR TITLE
Enable 'Edit Database Cell' when view is editable (#1756)

### PR DESCRIPTION
**Pull request by cristilupascu**
Fixed issue #1756.

### CODE REVIEW

**Comment by justinclift** [comment]
Thanks @cristilupascu.  Hopefully @MKleusberg or @mgrojo has time to review this before too long. :smile:

File: src/MainWindow.cpp [hunk 1]
```diff
@@ -1335,7 +1335,7 @@ void MainWindow::dataTableSelectionChanged(const QModelIndex& index)
     }
 
     bool editingAllowed = !db.readOnly() && (m_currentTabTableModel == m_browseTableModel) &&
-            (db.getObjectByName(currentlyBrowsedTableName())->type() == sqlb::Object::Types::Table);
+            m_browseTableModel->isEditable();
 
     // Don't allow editing of other objects than tables
```
Comment (mgrojo) [inline src/MainWindow.cpp]: Thank you very much! A simple change to fix this oversight, but it's demonstrating that you've carfully followed the code to pinpoint it. Congratulations!

Comment (cristilupascu) [reply src/MainWindow.cpp]: Thank you! It seemed simple and I really wanted that functionality, mainly for the JSON highlighter/editor.

**Comment by MKleusberg** [comment]
Going to merge this then :smile:  The fix looks correct at first glance and I'm fully trusting @mgrojo's review skills :wink:

If you ever feel like fixing more problems, please go for it @cristilupascu :+1: We also try to help wherever we can if new developers or just people interested in the code have questions.

**Comment by justinclift** [comment]
Should this be added to the `3.11.x` branch?

**Comment by MKleusberg** [comment]
Why not, it seems simple enough not to break anything :smile: I have just cherry-picked the commit.

**Comment by justinclift** [comment]
Cool.  Just added it to the tracking bug. :smile:

**Pull request merged**